pyccel/codegen/printing/ccode.py

```diff
@@ -693,20 +688,18 @@ def _print_NumpyRandint(self, expr):
 
     def _print_FunctionDef(self, expr):
 
+        if len(expr.results) > 1:
+            self._additional_args = expr.results
         body  = self._print(expr.body)
         decs  = [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.local_vars]
-        decs += [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.results]
+        if len(expr.results) <= 1 :
+            decs += [Declare(i.dtype, i) if isinstance(i, Variable) else FuncAddressDeclare(i) for i in expr.results]
         decs += [Declare(i.dtype, i) for i in self._additional_declare]
         decs  = '\n'.join(self._print(i) for i in decs)
         self._additional_declare.clear()
 
-        if len(expr.results) > 1:
-            self._additional_args = [a.clone(name = self._parser.get_new_name('tmp_ret'), is_pointer =True) for a in expr.results]
-            body += '\n'.join(self._print_Assign(Assign(a, b)) for  a, b in zip(self._additional_args, expr.results))
-            body += '\n' + self._print_Return(Return([BooleanTrue()]))
-
         sep = self._print(SeparatorComment(40))
-
+        self._additional_args = []
```

Comment:
This will not work if you have a function in a function

Reply:
I suggest initialising `self._additional_args` in `__init__`.
Then here you can do:
`self._additional_args.pop()`
This combined with the correction at line 692 should work

Reply:
Just to make it clear what do you mean by function in a function ?

Reply:
In python:
```
def f():
    def g():
        return 1, False
    if True:
        return g()
    else:
        return 2, True
```
In C:
```
int f(int* Out_1, bool* Out_2)
{
    int g(int* Out_3, bool* Out_4)
    {
        (*Out_3) = 1;
        (*Out_4) = 0;
        return 0;
    }
    if (1)
    {
        g(Out_1, Out_2);
        return 0;
    }
    else
    {
        (*Out_1) = 2;
        (*Out_2) = 1;
        return 0;
    }
}
```